types/stylelint/index.d.ts

```diff
@@ -275,26 +275,62 @@ declare namespace stylelint {
 		[key in `expected${string}` | `rejected${string}`]: RuleMessage;
 	};
 
-	type AutofixMessage = { expected: (actual: string, expected: string) => string };
+	type Message<T, method extends string, types extends Record<string, unknown[]>> = {
+		[key in `${method}`]: (
+			...args: T extends keyof types ? types[T] : T extends unknown[] ? T : never
+		) => string;
+	};
+
+	type ExpectedMessage<T> = Message<
+		T,
+		'expected',
+		{
+			autofix: [actual: string, expected: string];
+			pattern: [input: string, pattern: StringOrRegex];
+			max: [object: string, maximum: number];
+		}
+	>;
+
+	type RejectedMessage<T> = Message<T, 'rejected', {}>;
```

Comment:
[suggestion] IMO, these new types are too tricky for me, and they seem so hard to maintain in the future. Like the following examples, allowing some duplications makes the code more straightforward.

```typescript
type AutofixMessages = {
	expected: (actual: string, expected: string) => string;
};
type PatternMessages = {
	expected: (input: string, pattern: StringOrRegex) => string;
};
type MaximumMessages = {
	expected: (target: string, maximum: number) => string;
};
type RejectedMessages<T extends unknown[]> = {
	rejected: (...args: T) => string;
};
```

Reply:
Also, I think the `RejectedMessages` name is more accurate than `RejectedMessage` because they are objects that can have multiple message functions.

Reply:
@ybiquitous thanks for the review.
Sorry for the delay, Ill try to come up with something simpler when Ill have time.

This PR is not blocking anything.
I am planning to eventually generate the documentation automatically but that's not urgent.

Reply:
I have refactored the types.
It's now easier to maintain if you are not a TS expert.